### ISSUE TITLE
Incorrect max heap for G1GC

### ISSUE THREAD

**Issue by mpollmeier** [oracle/visualvm#127]
**Describe the bug**
When using G1GC the displayed heap max size is twice as high as it really is. 

**To Reproduce**
```
echo "public class Memory {
  public static void main(String[] args) throws Exception {
    System.out.println(Runtime.getRuntime().maxMemory());
    Thread.sleep(99999999);
  }
}" > Memory.java

javac Memory.java
```

Using the default ParallelGC: run the below, connect with visualvm, numbers are correct:
```
java -Xmx128m -XX:+UseParallelGC Memory
```
![screenshot_2019-03-05_13-49-46](https://user-images.githubusercontent.com/506752/53773099-e47deb80-3f4d-11e9-809a-2fe31d0a0377.png)

With G1GC: numbers are twice as high in visualvm (but correct on application stdout):
```
java -Xmx128m -XX:+UseG1GC Memory
```
![screenshot_2019-03-05_13-50-18](https://user-images.githubusercontent.com/506752/53773105-e778dc00-3f4d-11e9-9b06-ddb95e578c6e.png)

**Additional context**
Tested with the following jdks:
* oracle 1.8.0_144
* openjdk 1.8.0_202
* openjdk 11.0.1 (zulu11.2.+3 build)

**Comment by thurka** [comment]
This bug can be reproduce only if the memory is monitored via jvmstat (typically for local process). Monitoring via JMX (typically remote process) works fine.

**Comment by thurka** [comment]
[g1MonitoringSupport.hpp](http://hg.openjdk.java.net/jdk-updates/jdk11u/file/tip/src/hotspot/share/gc/g1/g1MonitoringSupport.hpp) describes G1 jvmstat counters. This bug is caused by the following paragraph:
```
*   Max Capacity
   For jstat, we set the max capacity of all spaces to heap_capacity,
   given that we don't always have a reasonable upper bound on how big
   each space can grow. For the memory pools, we make the max
   capacity undefined with the exception of the old memory pool for
   which we make the max capacity same as the max heap capacity.
```
Sum of max capacity of all spaces is overall max heap capacity for all other GC implementations. Unfortunately this is not the case for G1.

**Comment by mpollmeier** [comment]
Good to know that it works fine with JMX, thanks for looking into it! I suppose it's not blocking anyone from doing anything, just wanted to share the finding.

**Comment by jcbritobr** [comment]
Yep, I could observe this profiling a javafx application. I was thinking I calculated wrong the memory. -Xmx100m is configured

![image](https://user-images.githubusercontent.com/6579862/104776922-9a750c00-5759-11eb-9b5d-697cefa9aecc.png)
